src/wasm-type.h

```diff
@@ -377,13 +377,17 @@ struct Tuple {
   Tuple(std::initializer_list<Type> types) : types(types) { validate(); }
   Tuple(const TypeList& types) : types(types) { validate(); }
   Tuple(TypeList&& types) : types(std::move(types)) { validate(); }
+
+  // Allow copies when constructing.
+  Tuple(const Tuple& other) : types(other.types) { validate(); }
```

Comment:
Probably no need to validate here, since the other tuple is presumably already valid.